### PR TITLE
feat(pytest): add codefresh unit-test helper function and fix buildkite unit-tests

### PR DESCRIPTION
**Pull request by ericlaz**
## What does this PR do?
This PR adds support for Codefresh unit tests in order to pass the CI spec tests. Additionally, it adds support for Buildkite agent metadata variables.

Additionally, it updates the CI spec JSONs to follow the new standards.

## Checklist

- [X] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: ddtrace/ext/ci.py
Comment (romainkomorndatadog) [inline ddtrace/ext/ci.py:214]: Kinda scratching my head at this since dictionary comparisons in Python<3.6 don't care about order, but this test failure may be related to the ordering change?

https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/39916/workflows/d8ff6592-44e6-460c-a345-e7178033348f/jobs/2680890

File: ddtrace/ext/ci.py
Comment (gnufede) [inline ddtrace/ext/ci.py:324]: This one looks different than the rest

File: ddtrace/ext/ci.py
Comment (gnufede) [inline ddtrace/ext/ci.py:329]: Since you're calling several times `env.get("CF_BUILD_ID")`, why not store it's value to a variable for improved reuse?

Comment (ericlaz) [reply ddtrace/ext/ci.py:329]: Great suggestion, added!

File: ddtrace/ext/ci.py
Comment (ericlaz) [reply ddtrace/ext/ci.py:214]: Looks like it is passing now, I used `json.loads` to compare the dictionary instead. Thanks for the suggestion!

File: tests/tracer/fixtures/ci/bitbucket.json
Comment (ericlaz) [reply tests/tracer/fixtures/ci/bitbucket.json:8]: Good catch! Should be fixed in the next release of the CI-spec

File: ddtrace/ext/ci.py
Comment (romainkomorndatadog) [inline ddtrace/ext/ci.py:256]: Very small nit (just to show I actually read this PR): `BUILDKITE_AGENT_META_DATA_PREFIX` looks like a constant but it's just a variable in this function, so I'd make it lowercase.

File: ddtrace/ext/ci.py
Comment (romainkomorndatadog) [inline ddtrace/ext/ci.py:325]: Is this something other than `CF_BUILD_URL` specified in https://codefresh.io/docs/docs/pipelines/variables/ ?

If it is, is there a way we can use `CF_URL` instead of hardcoding `g.codefresh.io` ?

**Review by romainkomorndatadog — APPROVED** [review]
Looks good to me. Please do get a second approval from someone else (probably someone on the APM core reviewers list ).

File: releasenotes/notes/fix-codefresh-integration-563b8075e71881e4.yaml
Comment (majorgreys) [inline releasenotes/notes/fix-codefresh-integration-563b8075e71881e4.yaml:4]:
```suggestion
  - |
    CI visibility: Adds BITBUCKET_GIT_HTTP_ORIGIN environment variable
  - |
    CI visibility: Adds CodeFresh integration
```

**Review by majorgreys — COMMENTED** [review]
In general, my advice would be to split a PR like this into multiple feature PRs and a separate fix PR for any test changes. If you want to keep them all together, I would suggest splitting up the features in the release notes.

File: releasenotes/notes/fix-codefresh-integration-563b8075e71881e4.yaml
Comment (ericlaz) [reply releasenotes/notes/fix-codefresh-integration-563b8075e71881e4.yaml:4]: Good idea! I will split up the features in the release notes as suggested.

**Comment by juan-fernandez** [comment]
> In general, my advice would be to split a PR like this into multiple feature PRs and a separate fix PR for any test changes. If you want to keep them all together, I would suggest splitting up the features in the release notes.

@majorgreys thanks for the review!

That'd be hard, because the fixtures follow the logic in the extraction of metadata, so the tests would fail. In any case, for future PRs we can at least keep CI providers separate, to make the PR smaller 😄 . 

Other than this, do you see any other issue with the PR?

**Review by majorgreys — DISMISSED** [review]

**Review by mabdinur — APPROVED** [review]

**Review by romainkomorndatadog — DISMISSED** [review]

**Review by gnufede — APPROVED** [review]

File: releasenotes/notes/fix-codefresh-integration-563b8075e71881e4.yaml
Comment (mabdinur) [inline releasenotes/notes/fix-codefresh-integration-563b8075e71881e4.yaml:4]: Can we reword this release note. It's not immediately clear that this PR adds support for tracking the repository url via BITBUCKET_GIT_HTTP_ORIGIN environment variable.

File: ddtrace/ext/ci.py
Comment (mabdinur) [inline ddtrace/ext/ci.py:216]: Just curious why is an ordered no longer necessary?

File: releasenotes/notes/fix-codefresh-integration-563b8075e71881e4.yaml
Comment (ericlaz) [reply releasenotes/notes/fix-codefresh-integration-563b8075e71881e4.yaml:4]: Done, good catch!

File: ddtrace/ext/ci.py
Comment (ericlaz) [reply ddtrace/ext/ci.py:216]: Since we are using `json.loads` in order to compare both dictionaries JSON output strings, order does not matter when comparing them.

Relevant code snippet:
``py
b = '{"a": "d", "c": "a", "d": "f"}'
c = '{"c": "a", "a": "d", "d": "f"}'
print(json.loads(b) == json.loads(c)) # True on Python 2.7 and Python 3.10
``

File: ddtrace/ext/ci.py
Comment (gnufede) [inline ddtrace/ext/ci.py:255]:
```suggestion
    # Get all keys which start with BUILDKITE_AGENT_META_DATA_x
    node_label_list = []  # type: List[str]
```

File: ddtrace/ext/ci.py
Comment (gnufede) [inline ddtrace/ext/ci.py:424]:
```suggestion
    node_labels_list = []  # type:  List[str]
```

File: ddtrace/ext/ci.py
Comment (gnufede) [reply ddtrace/ext/ci.py:255]: You'll also need to add to the top imports:
```
from typing import List
```

**Review by majorgreys — APPROVED** [review]

**Pull request merged**
